src/lib/api/documents.ts

```diff
@@ -164,6 +163,21 @@ export async function process(
  */
 export async function cancel(id: number | string) {}
 
+export async function pending(fetch = globalThis.fetch): Promise<Pending[]> {
+  const endpoint = new URL("documents/pending/", BASE_API_URL);
+
+  // if this request fails for any reason, just return nothing
+  const resp = await fetch(endpoint, { credentials: "include" }).catch((e) => ({
+    ok: false,
+    json() {
+      return [];
+    },
+  }));
+
+  // if there's an error above, this will return an empty array
+  return resp.json();
```

Comment:
I could be mistaken, but this won't return an empty array if the request succeeds but returns a 400/500 error code, since fetch doesn't throw on non-200 responses. What about handling the request like this:

```suggestion
  try {
    const resp = await fetch(endpoint, { credentials: "include" });
    if (isErrorCode(resp.status)) throw new Error(resp.statusText);
    return resp.json();
  } catch {
    // if this request fails for any reason, just return an empty array
    return [];
  }
```